packages/@o3r/rules-engine/schematics/index.it.spec.ts

```diff
@@ -0,0 +1,120 @@
+const execAppOptions: ExecSyncOptions = {
+  cwd: appFolderPath,
+  stdio: 'inherit',
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  env: {...process.env, JEST_WORKER_ID: undefined, NODE_OPTIONS: ''}
+};
+const registry = 'http://localhost:4873';
+const o3rVersion = '999.0.0';
+
+/**
+ * Set up a local npm registry inside a docker image before the tests.
+ * Publish all the packages of the Otter monorepo on it.
+ * Can be accessed during the tests with url http://localhost:4873
+ */
+function setupLocalRegistry() {
+  let shouldHandleVerdaccio = false;
+
+  beforeAll(async () => {
+    try {
+      await getPidFromPort(4873);
+    } catch (ex) {
+      shouldHandleVerdaccio = true;
+      execSync('yarn verdaccio:start', {cwd: currentFolder, stdio: 'inherit'});
+      execSync('yarn verdaccio:publish', {cwd: currentFolder, stdio: 'inherit'});
+    }
+  });
+
+  afterAll(() => {
+    if (shouldHandleVerdaccio) {
+      execSync('yarn verdaccio:stop', {cwd: currentFolder, stdio: 'inherit'});
+    }
+  });
+}
+
+/**
+ * Setup a new application using Angular CLI
+ */
+function setupNewApp() {
+  beforeAll(() => {
+    const packageJson = JSON.parse(readFileSync(packageJsonPath).toString()) as PackageJson;
+    const angularVersion = minVersion(packageJson.devDependencies['@angular/core']).version;
+    // Create app with ng new
+    execSync(`npx rimraf it-tests/${appName}`, {cwd: parentFolderPath, stdio: 'inherit'});
+    if (!existsSync(itTestsFolderPath)) {
+      mkdirSync(itTestsFolderPath);
+    }
+    execSync(`npx --yes -p @angular/cli@${angularVersion} ng new ${appName} --style=scss --routing --defaults=true --skip-git --package-manager=yarn --skip-install`,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      {cwd: itTestsFolderPath, stdio: 'inherit', env: {...process.env, NODE_OPTIONS: ''}});
+
+    // Set config to target local registry
+    const o3rPackageJson: PackageJson & { packageManager?: string } = JSON.parse(fs.readFileSync(path.join(currentFolder, 'package.json')).toString());
+    const yarnVersion = o3rPackageJson?.packageManager?.split('@')?.[1] || '3.5.0';
+    execSync('yarn config set enableStrictSsl false', execAppOptions);
+    execSync(`yarn set version ${yarnVersion}`, execAppOptions);
+    execSync(`yarn config set npmScopes.ama-sdk.npmRegistryServer ${registry}`, execAppOptions);
+    execSync(`yarn config set npmScopes.o3r.npmRegistryServer ${registry}`, execAppOptions);
+    execSync('yarn config set unsafeHttpWhitelist localhost', execAppOptions);
+    execSync('yarn config set nodeLinker pnp', execAppOptions);
+    execSync('yarn config set enableMirror false', execAppOptions);
+    execSync(`yarn config set cacheFolder ${cacheFolderPath}`, execAppOptions);
+    if (existsSync(cacheFolderPath)) {
+      readdirSync(cacheFolderPath).forEach((fileName) => {
+        if (/^@(?:ama-sdk|ama-terasu|o3r)-/.test(fileName)) {
```

Comment:
could. be futur proof if you read this scope list from the workspace package.json's workspaces field

Reply:
updated